### ISSUE TITLE
"On" gets called twice 

### ISSUE THREAD

**Issue by miguelmolez** [garytee/homebridge-remote-ssh#4]
Hi,

This is really baffling me, but the "on" function seems to be getting called twice. Any idea how this could be happening?
Its causing the request to fail.

The "stop" works fine. Both work on executed locally

Thanks

[11/15/2022, 8:16:41 AM] [xmas] Set xmas to on
[11/15/2022, 8:16:41 AM] [xmas] Set xmas to on

    "accessories": [
        {
            "accessory": "SSH",
            "name": "xmas",
            "on": "catt -d SHIELD cast /xmas/xmas.mp4",
            "off": "catt -d 'SHIELD' stop",
            "ssh": {
                "user": "root",
                "host": "192.168.1.23",
                "port": 22,
                "password": "******"
            }
        }

**Comment by garytee** [comment]
Yeah, this app has a lot of issues. I'll put some time aside this week to update it but in the meantime I'd like to help you find a workaround. 

I tried recreating your issue and installed catt on my Mac. I then ran into completely separate issues which required me to create three scripts: cattOn.sh, cattOff.sh & cattStatus.sh

**cattOn.sh**

```
#!/bin/bash

osascript -e 'tell application "Terminal" to activate' -e 'tell application "Terminal" to do script "catt cast \"https://www.youtube.com/watch?v=dQw4w9WgXcQ\""'

echo "echo on" &>cattStatus.sh
```

**cattOff.sh**

```
#!/bin/bash

osascript -e 'tell application "Terminal" to activate' -e 'tell application "Terminal" to do script "catt stop"'

echo "echo off" &>cattStatus.sh
```

**cattStatus.sh**

```
echo off
```

**Homebridge Config**

```
{
    "accessory": "SSH",
    "name": "Raspberry Pi Cast Test",
    "on": "/Path/To/cattOn.sh",
    "off": "/Path/To/cattOff.sh",
    "state": "/Path/To/cattStatus.sh",
    "on_value": "on",
    "exact_match": false,
    "ssh": {
        "user": "username",
        "host": "usernames-computer.local",
        "port": 22,
        "password": "",
        "key": "/Path/To/.ssh/id_rsa"
    }
},
```

This is incredibly hacky but should work if the machine you're connecting to with SSH is a Mac. For whatever reason the scripts were not executing without applescript opening a terminal window so it's Mac only right now. If you have a Mac let me know if that works. If you have another OS let me know and I'll test it there.

Hopefully I can get this resolved without a workaround but figured I'd send you what worked for me until I get some time to properly update the repo.

**Comment by miguelmolez** [comment]
Thanks for the speedy response and the work you've put in to this so far! 
I'm  trying to SSH over to a Fedora VM. I have a MacBook, but would prefer it to be the VM which is at home and on all the time you see. I will have a little play with what you've put together, as I might still look at using it on my Mac for now.

**Comment by garytee** [comment]
Tested and working in a Fedora VM:

**cattOn.sh**
```
#!/bin/bash

# osascript -e 'tell application "Terminal" to activate' -e 'tell application "Terminal" to do script "catt cast \"https://www.youtube.com/watch?v=dQw4w9WgXcQ\""'

catt cast "https://www.youtube.com/watch?v=dQw4w9WgXcQ"

echo "echo on" &>cattStatus.sh
```

**cattOff.sh**
```
#!/bin/bash

# osascript -e 'tell application "Terminal" to activate' -e 'tell application "Terminal" to do script "catt stop"'

catt stop

echo "echo off" &>cattStatus.sh
```

Hopefully I have some time this weekend to give this repo an overhaul, but for now, this should work.

**Comment by miguelmolez** [comment]
Thanks for this!
I'll have a look again myself over the weekend. 
Cheers :)

**Comment by Z1mDMan** [comment]
It’s been a couple years, has there been any update to this issue? I experience this daily (with an automation) where the “on” gets called twice. I have considered the work-around however if that is also calling twice, then it doesn’t make sense for me to try it. Please advise if the work-around works and if this is something that can be addressed in the coding.

Thanks!
